code/turf/turf_flock.dm

```diff
@@ -126,13 +139,121 @@
 	src.health = initial(health)
 	src.name = initial(name)
 	src.desc = initial(desc)
+	if(isnull(src.group)) checknearby() //check for groups to join
+	for(var/obj/flock_structure/f in get_turf(src))
+		if(f.usesgroups)
+			f.group = src.group
+			f.group.addstructure(f)
 
 /turf/simulated/floor/feather/broken
 	name = "weird broken floor"
 	desc = "Disco's dead, baby."
 	icon_state = "floor-broken"
 	broken = 1
 
+////////////////////////////////////////////////////////////////////////////////////////
+//start of flocktilegroup stuff
+
+/turf/simulated/floor/feather/proc/initializegroup() //make a new group
+	group = new/datum/flock_tile_group
+	group.addtile(src)
+
+/turf/simulated/floor/feather/proc/checknearby(var/newgroup = 0)//handles merging groups
+	var/list/tiles = list() //list of tile groups found
+	var/datum/flock_tile_group/largestgroup = null //largest group
+	var/max_group_size = 0
+	for(var/turf/simulated/floor/feather/F in getneighbours(src))//check for nearby flocktiles
+		if(F.group && !F.broken)
+			if(F.group.size > max_group_size)
+				max_group_size = F.group.size
+				largestgroup = F.group
+			tiles |= F.group
+	if(tiles.len == 1)
+		src.group = tiles[1] //set it to the group found.
+		src.group.addtile(src)
+	else if(tiles.len > 1) //if there is more then one, then join the largest (add merging functionality here later)
+		for(var/datum/flock_tile_group/FUCK in tiles)
+			if(FUCK == largestgroup) continue
+			largestgroup.powergen += FUCK.powergen
+			largestgroup.poweruse += FUCK.poweruse
+			for(var/turf/simulated/floor/feather/F in FUCK.members)
+				F.group = largestgroup
+				largestgroup.addtile(F)
+			for(var/obj/flock_structure/f in FUCK.connected)
+				f.group = largestgroup
+				largestgroup.addstructure(f)
+			qdel(FUCK)
+		src.group = largestgroup
+		largestgroup.addtile(src)
+
+	else
+		if(newgroup)
+			src.initializegroup()
+		else return null
+
+/turf/simulated/floor/feather/proc/splitgroup()
+	var/list/groups = list() //list of tile groups found
+	var/count = 0 //how many tiles
+	var/turf/simulated/floor/feather/F
```

Comment:
what the heck is going on with this cursed reused variable hellhole please no

Reply:
only the F var was reused in this proc(unless you're reffering to other procs). pushed a change.